### PR TITLE
vim-patch:8.1.2094: the fileio.c file is too big

### PR DESCRIPTION
**Pull request by lewis6991**
Problem:    The fileio.c file is too big.
Solution:   Move buf_write() to bufwrite.c. (Yegappan Lakshmanan, closes vim/vim#4990)

https://github.com/vim/vim/commit/c079f0fed1c16495d726d616c5362edc04742a0d

Co-authored-by: Yegappan Lakshmanan <yegappan@yahoo.com>

### CODE REVIEW

**Comment by clason** [comment]
Yes. Yes, it is.

File: src/nvim/fileio.h [hunk 1]
```diff
@@ -20,6 +20,28 @@
...
+#define CONV_RESTLEN 30
+
+#define BUFSIZE         8192    // size of normal write buffer
+
+// We have to guess how much a sequence of bytes may expand when converting
+// with iconv() to be able to allocate a buffer.
+#define ICONV_MULT 8
+
 #ifdef INCLUDE_GENERATED_DECLARATIONS
 // Events for autocommands
```
Comment (zeertzjq) [inline src/nvim/fileio.h:46]:
```suggestion
```

File: src/nvim/bufwrite.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#ifndef NVIM_BUFWRITE_H
+#define NVIM_BUFWRITE_H
+
+#include "nvim/buffer_defs.h"
+#include "nvim/ex_cmds_defs.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+// Events for autocommands
```
Comment (zeertzjq) [inline src/nvim/bufwrite.h:8]:
```suggestion
```

**Comment by zeertzjq** [comment]
It should be `vim/vim#4990` in commit message

**Comment by zeertzjq** [comment]
Build failure on Windows because of `sys/` headers
Missing:
- `filemess()` message overwrite
- `readfile()` message overwrite
- `BUFSIZE` to `WRITEBUFSIZE` rename

**Comment by lewis6991** [comment]
`BUFSIZE` is also used to read files.

**Comment by zeertzjq** [comment]
> `BUFSIZE` is also used to read files.

But why does that prevent it from being named `WRITEBUFSIZE`? As long as it is indeed used for writing files, naming it `WRITEBUFSIZE` makes sense, even if it's also used for other purposes.

**Review by zeertzjq — APPROVED** [review]
The order of some functions looks a bit strange. I pushed a commit to reorder them.

**Pull request merged**